duecredit/collector.py

```diff
@@ -195,20 +197,62 @@ def cite(self, entry, **kwargs):
 
         return citation
 
+    @staticmethod
+    def _args_match_conditions(conditions, *fargs, **fkwargs):
+        """Helper to identify when to trigger citation given parameters to the function call
+        """
+        hit_default = False
+        for (arg, kwarg), values in iteritems(conditions):
+            if len(fargs) > arg:
+                return fargs[arg] in values
+            if kwarg in fkwargs:
+                return fkwargs[kwarg] in values
+            if 'DC_DEFAULT' in values:
+                hit_default = True
+        return hit_default
+
     @never_fail
     @borrowdoc(Citation, "__init__", replace="PLUGDOCSTRING")
     def dcite(self, *args, **kwargs):
         """Decorator for references.  PLUGDOCSTRING
 
+        Parameters
+        ----------
+        conditions: dict, optional
+          If reference should be cited whenever parameters to the function call
+          satisfy given values.  Use "DC_DEFAULT" keyword as a value to depict default
+          value (e.g. if no explicit value was provided for that positional or keyword
+          argument).
+
+          Notes:
+            first matching argument (or e.g. absence of it, given presence of
+            "DC_DEFAULT") would result in match and order of items within conditions is not
+            guaranteed unless you use some OrderedDict.  Overall matching multiple cases
+            is not fully tested and we would appreciate a feedback and use-cases
+
         Examples
         --------
 
-        @due.dcite('XXX00', description="Provides an answer for meaningless existence")
-        def purpose_of_life():
-            return None
+        >>> from duecredit import due
+        >>> @due.dcite('XXX00', description="Provides an answer for meaningless existence")
+        ... def purpose_of_life():
+        ...     return None
+
+        Conditional citation given argument to the function
+
+        >>> @due.dcite('XXX00', description="Relief through the movement",
+        ...            conditions={(1, 'method'): {'purge', 'DC_DEFAULT'}})
```

Comment:
great point!  for ordering let's just rely on people using OrdereDict.  Here it is an "index" for that argument if used "positionally". i.e. here "1" is because it is after "x" argument (which has index 0).  I have added to the docstring description